File: app/src/main/java/uk/co/agfsoft/skyweather/volley/requests/GetByNameRequest.java
package uk.co.agfsoft.skyweather.volley.requests;

import com.android.volley.Request;

import java.util.HashMap;
import java.util.Map;

import javax.inject.Inject;

import uk.co.agfsoft.skyweather.BuildConfig;
import uk.co.agfsoft.skyweather.model.FindCitiesResponse;
import uk.co.agfsoft.skyweather.model.RawWeatherCity;
import uk.co.agfsoft.skyweather.volley.BaseVolleyRequest;



public class GetByNameRequest extends BaseVolleyRequest<FindCitiesResponse> {

    private static final String Q = "q";
    private static final String APPID = "APPID";
    public static final String TYPE = "type";
    public static final String LIKE = "like";

    private Map<String, String> queryParams;

    @Inject
    public GetByNameRequest() {
    }

    @Override
    protected String getUrl() {
        return BuildConfig.OPEN_WEATHER_FIND_URL;
    }

    @Override
    protected int getMethod() {
        return Request.Method.GET;
    }

    @Override
    protected Class getResponseClass() {
        return FindCitiesResponse.class;
    }


    @Override
    protected Map<String, String> getQueryParams() {
        return queryParams;
    }

    public void setQuery(String cityName) {
        queryParams = new HashMap<>(3);
        queryParams.put(Q, cityName);
        queryParams.put(TYPE, LIKE);
        queryParams.put(APPID, BuildConfig.OPEN_WEATHER_API_KEY);
    }
}

File: app/src/main/kotlin/volley/requests/GetByNameRequest.kt
package uk.co.agfsoft.skyweather.volley.requests

import com.android.volley.Request

import java.util.HashMap

import javax.inject.Inject

import uk.co.agfsoft.skyweather.BuildConfig
import model.FindCitiesResponse
import uk.co.agfsoft.skyweather.volley.BaseVolleyRequest


class GetByNameRequest @Inject
constructor() : BaseVolleyRequest<FindCitiesResponse>() {

    private var queryParams: MutableMap<String, String>? = null

    protected override val method: Int
        get() = Request.Method.GET

    protected override val responseClass: Class<*>
        get() = FindCitiesResponse::class.java

    override fun getUrl(): String {
        return BuildConfig.OPEN_WEATHER_FIND_URL
    }


    override fun getQueryParams(): Map<String, String>? {
        return queryParams
    }

    fun setQuery(cityName: String) {
        queryParams = HashMap(3)
        queryParams!![Q] = cityName
        queryParams!![TYPE] = LIKE
        queryParams!![APPID] = BuildConfig.OPEN_WEATHER_API_KEY
    }

    companion object {

        private val Q = "q"
        private val APPID = "APPID"
        val TYPE = "type"
        val LIKE = "like"
    }
}
